libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -571,32 +585,48 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 			else
 				args.emplace_back(convert(*arguments[i], *parameterTypes[i]).commaSeparatedList());
 
-		if (auto identifier = dynamic_cast<Identifier const*>(&_functionCall.expression()))
+		FunctionDefinition const* functionDef;
+		if (auto memberAccess = dynamic_cast<MemberAccess const*>(&_functionCall.expression()))
+		{
+			solUnimplementedAssert(!functionType->bound(), "Internal calls to bound library functions are not yet implemented");
+
+			functionDef = dynamic_cast<FunctionDefinition const*>(memberAccess->annotation().referencedDeclaration);
+			solAssert(functionDef, "Libraries cannot have state variables");
+			solAssert(!functionDef->virtualSemantics(), "Libraries can neither inherit nor be inherited");
```

Comment:
Please remove this assertion. Internal function calls with "qualified access" never use virtual resolution, but the function can very well be virtual.

Maybe we should add an assertion `functionDef.isImplemented` instead.

Reply:
OK. Replaced with `isImplemented()`.

Actually I made this check when I still intended this work to only with libraries (there was an assert for that just above the `switch`). But yeah, now it can be also a contract function and those can be virtual.